server/room_db.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/golang/protobuf/ptypes"
+
+	pb "github.com/googleapis/gapic-showcase/server/genproto"
+	"google.golang.org/genproto/protobuf/field_mask"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func NewRoomDb() RoomDb {
+	return &roomDb{
+		uid:   &uniqID{},
+		token: NewTokenGenerator(),
+
+		mu:    sync.Mutex{},
+		keys:  map[string]int{},
+		rooms: []roomEntry{},
```

Comment:
`mu` and `rooms` are essentially set to zero value. We can just elide the initialization.